app/main.py

```diff
@@ -2,5 +1,19 @@
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    saved_results = {}
+
+    @wraps(func)
+    def wrapper(*args: Any) -> Any:
+        if args in saved_results:
+            print("Getting from cache")
+            return saved_results[args]
+        else:
+            result = func(*args)
+            saved_results[args] = result
```

Comment:
Similar to line 10, storing `args` directly as a key can cause issues if any argument is mutable. Ensure `args` is converted to a hashable type before using it as a key in `saved_results`.